### ISSUE TITLE
Plancherel's sum, integral of cabs(f)^2, leads to Lisp error "The value 1 is not of type LIST"

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#2677]
Imported from SourceForge on 2024-07-06 17:15:58
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2018-05-16 04:44:16
Original: https://sourceforge.net/p/maxima/bugs/3429

---

From the mailing list 2011-05-26, "Help with Plancherel's theorem" by dufault at hda.com.

Copied verbatim:
~~~~
====
declare ( x,[real,scalar], k,[real,scalar], a,real, b,real, c,real)$
assume( a>0, b>0, c>0 )$

/* Gaussian. */
gauss(x,a,b,c):=a*%e^(-((x-b)^2)/(2*c^2))$

/* Fourier transform. */
FT(fx,x,k):=integrate(fx*%e^(-2 * %i * %pi * k * x), x, minf, inf)$

/* An even function (shifted gaussian) + odd function (two shifted gaussians, one negative): */
feven:gauss(x,1,0,1/sqrt(2))$
fodd:gauss(x - 1, a, b, c) - gauss(x + 1, a, b, c)$
f:feven+fodd$

/* "Plancherel's sum" - just integrate the square of the modulus over all values. */
planch(f,v):=factor(integrate(cabs(f)^2,v,minf,inf))$

/* I expect these to all be 0. */
planch(feven,x) - planch(FT(feven,x,k),k);  /* OK. */
planch(fodd,x)  - planch(FT(fodd,x,k),k);   /* OK. */
planch(f,x)     - planch(FT(f,x,k),k);      /* Huh? */
====

...
(%i9)      planch(f, v) := factor(integrate(cabs (f), v, minf, inf))
(%i10)           planch(feven, x) - planch(FT(feven, x, k), k)
(%o10)                                 0
(%i11)            planch(fodd, x) - planch(FT(fodd, x, k), k)
(%o11)                                 0
(%i12)               planch(f, x) - planch(FT(f, x, k), k)
Maxima encountered a Lisp error:

 The value 1 is not of type LIST.
~~~~

I find that `planch(f, x)` asks asksign questions. For some combinations of responses, eventually Maxima comes to a result (I didn't check if it's correct). For asksign responses `z`, `z`, and `z`, I get the error "1 is not of type LIST". 

A stack trace shows that it's somewhere in `limit`. Tracing limit, I see that the offending call is `limit((a*%e^-(((-b) + 1/x + 1)^2/(2*c^2)) - a*%e^-(((-b) + 1/x - 1)^2/(2*c^2)) - %e^-(1/x^2))^2/x, x, 0, minus)`.  When I try that immediately following the previous error, then I get the same "1 is not of type LIST" error. But when I try that in a fresh session, I get a couple of asksign questions and then a `limit` noun expression -- I guess there must be some global state hanging around after the first error.

A backtrace shows that the error arises in function CPA1.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 17:16:00
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2022-02-28 00:11:01
Original: https://sourceforge.net/p/maxima/bugs/3429/#45b7

---

The Lisp error seems to have gone away (not observed in post-5.45), and the call to `limit` shown above returns 0 after a couple of asksigns (dunno if it's correct). But `planch(f, x)` doesn't seem to return 0 as expected; instead a result numerically equal to about 1.36 is returned for some combinations of asksign responses. I didn't look into it carefully.

If the Lisp error has gone away, it's probably appropriate to close this report (adding one or more tests for it), and open a new report for incorrect results from limit or simplification, if any.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 17:16:03
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2022-02-28 09:11:42
Original: https://sourceforge.net/p/maxima/bugs/3429/#a8f6

---

git bisect shows the Lisp error was fixed by commit 35060b1. Closing this report as fixed.

There is some other strangeness I've encountered while researching this; I'll open a new bug for that.
